src/neo/SmartContract/Native/Tokens/GasToken.cs

```diff
@@ -39,7 +39,7 @@ protected override bool OnPersist(ApplicationEngine engine)
         {
             if (!base.OnPersist(engine)) return false;
             foreach (Transaction tx in engine.Snapshot.PersistingBlock.Transactions)
-                Burn(engine, tx.Sender, tx.SystemFee + tx.NetworkFee);
+                Burn(engine, tx);
```

Comment:
I think this is incorrect. The event isn't triggered by the transaction. It is triggered by the block. We shouldn't change this part.

Reply:
I deal with the gas burn case in the [RpcNep5Tracker ](https://github.com/neo-project/neo-modules/pull/181 )class, which wont modify codes in neo repo.